serve.go

```diff
@@ -24,26 +29,52 @@ func (s *Server) Run() error {
 // It also generates the OpenAPI spec and outputs it to a file, the UI, and a handler (if enabled).
 func (s *Server) RunTLS(certFile, keyFile string) error {
 	s.isTLS = true
-
-	s.setup()
-	return s.Server.ListenAndServeTLS(certFile, keyFile)
+	if err := s.setup(); err != nil {
+		return err
+	}
+	return s.Server.ServeTLS(s.listener, certFile, keyFile)
 }
 
-func (s *Server) setup() {
+func (s *Server) setup() error {
+	if err := s.setupDefaultListener(); err != nil {
+		return err
+	}
 	go s.OutputOpenAPISpec()
 	s.printStartupMessage()
 
 	s.Server.Handler = s.Mux
 	if s.corsMiddleware != nil {
 		s.Server.Handler = s.corsMiddleware(s.Server.Handler)
 	}
+
+	s.OpenApiSpec.Servers = append(s.OpenApiSpec.Servers, &openapi3.Server{
```

Comment:
Ah changing this causes some problems.

The issue is that we allow users to output the spec by itself. Would you mind putting it back to where it was before the PR. I'll open an issue to refactor it. 

My only thought is that how to do we know if the user is intending to be tls when just outputting the spec 🤷 we don't know that til start time. It probably doesn't matter honestly.